pandas/conftest.py

```diff
@@ -953,3 +954,76 @@ def __len__(self):
             return self._data.__len__()
 
     return TestNonDictMapping
+
+
+indices_dict = {
+    "unicode": tm.makeUnicodeIndex(100),
+    "string": tm.makeStringIndex(100),
+    "datetime": tm.makeDateIndex(100),
+    "datetime-tz": tm.makeDateIndex(100, tz="US/Pacific"),
+    "period": tm.makePeriodIndex(100),
+    "timedelta": tm.makeTimedeltaIndex(100),
+    "int": tm.makeIntIndex(100),
+    "uint": tm.makeUIntIndex(100),
+    "range": tm.makeRangeIndex(100),
+    "float": tm.makeFloatIndex(100),
+    "bool": tm.makeBoolIndex(2),
+    "categorical": tm.makeCategoricalIndex(100),
+    "interval": tm.makeIntervalIndex(100),
+    "empty": Index([]),
+    "tuples": MultiIndex.from_tuples(zip(["foo", "bar", "baz"], [1, 2, 3])),
+    "repeats": Index([0, 0, 1, 1, 2, 2]),
+}
+
+
+@pytest.fixture(params=indices_dict.keys())
+def indices(request):
+    # copy to avoid mutation, e.g. setting .name
+    return indices_dict[request.param].copy()
+
+
+def _create_series(index):
+    """ Helper for the _series dict """
+    data = np.random.randn(len(index))
+    return pd.Series(data, index=index, name=index.name)
+
+
+_series = {
+    f"series-with-{id_}-index": _create_series(index)
+    for id_, index in indices_dict.items()
+}
+
+
+def _create_narrow_series(dtype):
+    """ Helper for the _narrow_series dict """
+    index = indices_dict["int"].copy()
+    size = len(index)
+    if np.issubdtype(dtype, np.floating):
+        data = np.random.choice(size, size=size, replace=False)
+    elif np.issubdtype(dtype, np.integer):
+        data = np.random.randn(size)
+    else:
+        raise ValueError(f"Received an unexpected data_dtype: {dtype}")
+    return pd.Series(data.astype(dtype), index=index, name="a")
```

Comment:
do we not have something like this in pandas._testing that we can reuse here?

Reply:
I'm using `tm.makeFloatSeries` now which makes it much more concise ✨
I hope the differing index doesn't create any issues further down the road though 🤞